Commands/InteractionCommands/SecurityActionInteractions.cs

```diff
@@ -0,0 +1,94 @@
+namespace Cliptok.Commands.InteractionCommands
+{
+    public class SecurityActionInteractions : ApplicationCommandModule
+    {
+        [SlashCommand("pausedms", "Temporarily pause DMs between server members.", defaultPermission: false)]
+        [HomeServer, SlashRequireHomeserverPerm(ServerPermLevel.Moderator), SlashCommandPermissions(Permissions.ModerateMembers)]
+        public async Task SlashPauseDMs(InteractionContext ctx, [Option("time", "The amount of time to pause DMs for. Cannot be greater than 24 hours.")] string time)
+        {
+            // need to make our own api calls because D#+ can't do this natively?
+
+            // parse time from message
+            DateTime t = HumanDateParser.HumanDateParser.Parse(time);
+            if (t <= DateTime.Now)
+            {
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} Time can't be in the past!");
+                return;
+            }
+            if (t > DateTime.Now.AddHours(24))
+            {
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} Time can't be greater than 24 hours!");
+                return;
+            }
+            var dmsDisabledUntil = t.ToUniversalTime().ToString("o");
+            
+            // get current security actions to avoid unintentionally resetting invites_disabled_until
+            var currentActions = await SecurityActionHelpers.GetCurrentSecurityActions(ctx.Guild.Id);
+            JToken invitesDisabledUntil;
+            if (currentActions is null || !currentActions.HasValues)
+                invitesDisabledUntil = null;
+            else
+                invitesDisabledUntil = currentActions["invites_disabled_until"];
+            
+            // create json body
+            var newSecurityActions = JsonConvert.SerializeObject(new
+            {
+                invites_disabled_until = invitesDisabledUntil,
+                dms_disabled_until = dmsDisabledUntil,
+            });
+
+            // set actions
+            var setActionsResponse = await SecurityActionHelpers.SetCurrentSecurityActions(ctx.Guild.Id, newSecurityActions);
+
+            // respond
+            if (setActionsResponse.IsSuccessStatusCode)
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Success} Successfully paused DMs for **{TimeHelpers.TimeToPrettyFormat(t.Subtract(ctx.Interaction.CreationTimestamp.DateTime.ToLocalTime()), false)}**!");
```

Comment:
Why not Discord timestamps? PrettyFormat has a bunch of issues with time skew and I'm trying not to use it too much.
```suggestion
                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Success} Successfully paused DMs until <t:{TimeHelpers.ToUnixTimestamp(t)}>!");
```

Reply:
Good question! The first thing that came to mind for me was how we show times in warnings and stuff, which uses PrettyFormat. Discord timestamps are a better idea!